lib/srv/desktop/tdp/conn.go

```diff
@@ -117,3 +118,27 @@ func (c *Conn) LocalAddr() net.Addr {
 func (c *Conn) RemoteAddr() net.Addr {
 	return c.remoteAddr
 }
+
+// IsNonFatalErr returns whether or not an error arising from
+// the tdp package should be interpreted as fatal or non-fatal
+// for an ongoing TDP connection.
+func IsNonFatalErr(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	return errors.Is(err, trace.LimitExceeded(clipDataMaxLenErrMsg)) ||
```

Comment:
This is a pretty inefficient check, as you have to construct (up to) 4 errors for comparison, and each of the 4 `LimitExceeded` calls is going to compute the current stack trace.

WDYT about makign these sentinel errors at the package level?

```
var (
  mfaLimitExceededErr       = trace.LimitExceeded("asdfjkl")
  stringLenLimitExceededErr = trace.LimitExceeded("jkllp")
```

And then just checking `errors.Is` against those here.

Reply:
77cc8fc8dde8178d9706aaa1e23458a89f982b1d